src/main/java/com/parkinglot/SmartParkingBoy.java

```diff
@@ -0,0 +1,30 @@
+package com.parkinglot;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+
+public class SmartParkingBoy extends ParkingBoy{
+    public SmartParkingBoy(ParkingLot parkingLot) {
+        super(parkingLot);
+    }
+
+    public SmartParkingBoy(List<ParkingLot> parkingLots){
+        super(parkingLots);
+
+    }
+
+    @Override
+    public Ticket park(Car car) throws UnavailableSlotException {
+        Optional<ParkingLot> bestParkingLot = parkingLots
+                .stream()
+                .max(Comparator.comparing(parkingLot -> parkingLot.getCapacity() - parkingLot.getTicketCarMap().size()));
```

Comment:
maybe add a function of getRemainingSpace? To avoid having to know the capacity and size